### ISSUE TITLE
Submission: <46: Iris ML Predictor>

### ISSUE THREAD

**Issue by Diana-Cornescu** [UBC-MDS/data-analysis-review-2025#56]
Submitting authors: @suryashch @hoomanesteki @bbarafat

Repository: https://github.com/hoomanesteki/iris-ml-predictor
Report link: https://github.com/hoomanesteki/iris-ml-predictor/blob/main/reports/iris_report.pdf
Abstract/executive summary: This project investigates the capability of a machine learning model to classify iris flowers into one of the three species, namely, Setosa, Versicolor, and Virginica, based on four numerical measurements of the petals and sepals. The work is for the purpose of building a reproducible workflow that follows the syllabus of DSCI 522 (Data Science Workflows) course in UBC’s Master of Data Science program.

Editor: @skysheng7
Reviewer: @Diana-Cornescu 

I agree to abide by [MDS's Code of Conduct](https://ubc-mds.github.io/resources_pages/code_of_conduct/) during the review process and in maintaining my package should it be accepted.

**Comment by Diana-Cornescu** [comment]
## Data analysis review checklist

### Reviewer: <@Diana-Cornescu> 

### Conflict of interest

- [x] As the reviewer I confirm that I have no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [MDS code of conduct](https://ubc-mds.github.io/resources_pages/code_of_conduct/).

### General checks

- [x] **Repository:** Is the source code for this data analysis available? Is the repository well organized and easy to navigate?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?

### Documentation

- [x] **Installation instructions:** Is there a clearly stated list of dependencies? 
- [x] **Example usage:** Do the authors include examples of how to use the software to reproduce the data analysis?
- [x] **Functionality documentation:** Is the core functionality of the data analysis software documented to a satisfactory level?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Code quality
- [x] **Readability:** Are scripts, functions, objects, etc., well named? Is it relatively easy to understand the code?
- [x] **Style guidelides:** Does the code adhere to well known language style guides?
- [x] **Modularity:** Is the code suitably abstracted into scripts and functions?
- [x] **Tests:** Are there automated tests or manual steps described so that the function of the software can be verified? Are they of sufficient quality to ensure software robsutness?

### Reproducibility
 
- [x] **Data:** Is the raw data archived somewhere? Is it accessible?
- [x] **Computational methods:** Is all the source code required for the data analysis available?
- [x] **Conditions:** Is there a record of the necessary conditions (software dependencies) needed to reproduce the analysis? Does there exist an easy way to obtain the computational environment needed to reproduce the analysis?
- [x] **Automation:** Can someone other than the authors easily reproduce the entire data analysis?

### Analysis report

- [ ] **Authors:** Does the report include a list of authors with their affiliations?
- [x] **What is the question:** Do the authors clearly state the research question being asked?
- [ ] **Importance:** Do the authors clearly state the importance for this research question?
- [x] **Background**: Do the authors provide sufficient background information so that readers can understand the report?
- [x] **Methods:** Do the authors clearly describe and justify the methodology used in the data analysis? Do the authors communicate any assumptions or limitations of their methodologies?
- [x] **Results:** Do the authors clearly communicate their findings through writing, tables and figures?
- [x] **Conclusions:** Are the conclusions presented by the authors correct? 
- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?
- [x] **Writing quality:** Is the writing of good quality, concise, engaging? 

### Estimated hours spent reviewing:  ~2hrs

### Review Comments: 

Please provide more detailed feedback here on what was done particularly well, and what could be improved. It is especially important to elaborate on items that you were not able to check off in the list above.

- {**Community guidelines**}: Thorough and well-outlined steps to make a PR. However, no guidelines for third parties on how to report issues and minimal mention on how to "seek support".
- {**Documentation**}: Would advise against having the direct report results within the README. As (with a new dataset or an updated script) values such as "150 sample" or " accuracy of 86%" might change & require manual updating - which might be forgotten. 
- {**License**}: Good attribution mentions throughout the report and in the repo, but I believe an important place to have it is also within the License!
- {**Reproducibility**}: Well segmented code scripts. Analysis can be run from start to finish without any errors. 
	- However: missing step to actually `quarto render` .qmd file into the final report. Aka only mentions the current location of where reports can be found - but not how to update or recreate report.
	- (Extra) could be more explicit in the README that users must be in the root terminal directory in order to run the scripts.
	- (Extra) packages are semi-pinned in environment.yml file. Currently doesn't cause any issues but might in the future. Fully pinning the package versions `#.#.#` (vs `#.#.*`) would prevent this.
- {**Report**}: Concise and easy to read.
	- {Method}: Good overview of methodology - if time permits, it would be good to further expand on the default arguments that the model might be using (given no hyperparameter optimization was) and more discussion about limitation of DT. 
	- {Importance}: While it might be difficult due to the nature of the data set - ideally the "Importance" element of the report would be further expanded upon, beyond how it is mainly used to teach machine learning.
	- {Results}: Including the tabled results of DummyClassifer (performance metrics) along side the provided Decision Tree metrics would give more context and allow for better comparison between the DT and baseline. 
	- {References}: While DOI (link) cannot be provide for the Scikit-Learn reference (_JMLR_ journal quirk) - one can be added for the Iris Dataset.

_Extra thoughts and comments:_  I really liked the section of  `Future Work`  and what can be done to further expand the project. I also found your EDA and the Summary/Introduction to be well-presented and insightful. All in all, good job guys!  😊

#### Attribution

This was derived from the [JOSE review checklist](https://openjournals.readthedocs.io/en/jose/review_checklist.html) and the ROpenSci review checklist.v
